debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleValueConverters.java

```diff
@@ -235,21 +235,26 @@ protected Object convertDecimal(Column column, Field fieldDefn, Object data) {
             }
         }
 
+        // adjust scale to column's scale if the column's scale is larger than the one from
+        // the value (e.g. 4.4444 -> 4.444400)
+        if (data instanceof BigDecimal) {
+            data = withScaleAdjustedIfNeeded(column, (BigDecimal) data);
+        }
+
         return super.convertDecimal(column, fieldDefn, data);
     }
 
-    @Override
-    protected Object convertNumeric(Column column, Field fieldDefn, Object data) {
-        if (data instanceof NUMBER) {
-            try {
-                data = ((NUMBER)data).bigDecimalValue();
-            }
-            catch (SQLException e) {
-                throw new RuntimeException("Couldn't convert value for column " + column.name(), e);
-            }
+    private BigDecimal withScaleAdjustedIfNeeded(Column column, BigDecimal data) {
```

Comment:
The same code is present in postgres connector - maybe this is something that is common for all connectors https://github.com/debezium/debezium/blob/master/debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java#L346